### PR TITLE
DependencyInstaller.sh Fixes

### PR DESCRIPTION
**Pull request by rovinski**
Some small fixes and cleanups of the dependency installer script

* Check for and install pcre2 as a swig dependency (not all systems may have a suitable version)
* Eliminate some spurious warnings about "file not found" when grepping files that don't exist
* Formatting
* If installing to a non-default location, emit a sourceable script to set up the environment
* Convert relative paths to absolute when passing to installers

### CODE REVIEW

File: etc/DependencyInstaller.sh
Comment (oharboe) [inline etc/DependencyInstaller.sh:136]: Does env.sh and setup_env.sh in ORFS need to be updated?

**Review by oharboe — CHANGES_REQUESTED** [review]
Does env.sh and setup_env.sh in ORFS need to be updated?

File: etc/DependencyInstaller.sh
Comment (rovinski) [reply etc/DependencyInstaller.sh:136]: "need to be" - no. This is just a convenience addition for installations making use of `-prefix`. It doesn't change any behavior with current scripts, although after merging they could make use if they wanted to.

**Comment by rovinski** [comment]
@vvbandeira Any ideas about the failures? The all seem to be related to 
```
[ 43%] Linking CXX executable TestWorker
cd /OpenROAD/build/src/dst/test/cpp && /home/openroad-deps/bin/cmake -E cmake_link_script CMakeFiles/TestWorker.dir/link.txt --verbose=1
/usr/bin/clang++ -O3 -DNDEBUG CMakeFiles/TestWorker.dir/TestWorker.cc.o CMakeFiles/TestWorker.dir/stubs.cpp.o -o TestWorker  ../../dst.a /usr/lib64/libtcl8.6.so ../../../utl/utl.a ../../../utl/libutl_lib.a /home/openroad-deps/lib64/libspdlog.a /OpenROAD/src/sta/app/libOpenSTA.a /usr/lib64/libtcl8.6.so -pthread /usr/lib64/libtclreadline.so -ltcl /usr/lib64/libz.so /home/openroad-deps/lib/libboost_serialization.a /home/openroad-deps/lib/libboost_system.a /home/openroad-deps/lib/libboost_thread.a -lpthread 
/usr/bin/ld: cannot find -ltcl
clang-14: error: linker command failed with exit code 1 (use -v to see invocation)

which I don't believe is related to this change.
```

File: etc/DependencyInstaller.sh
Comment (oharboe) [reply etc/DependencyInstaller.sh:136]: So update env.sh and dev_env.sh to use a single source of truth instead of duplicating how to set up paths?

**Comment by rovinski** [comment]
Confirmed - even without this PR those tests are failing in the same way. @vvbandeira

File: etc/DependencyInstaller.sh
Comment (rovinski) [reply etc/DependencyInstaller.sh:136]: Sure

File: etc/DependencyInstaller.sh
Comment (maliberty) [inline etc/DependencyInstaller.sh:60]: Why is there a new dependency in this PR?

File: etc/DependencyInstaller.sh
Comment (maliberty) [reply etc/DependencyInstaller.sh:136]: @rovinski you asked for a review but this item seems unresolved.  Is there more to do yet?

File: etc/DependencyInstaller.sh
Comment (rovinski) [reply etc/DependencyInstaller.sh:60]: On some systems pcre2 isn't installed, pcre1 is. The SWIG install breaks if it can't find pcre2.

File: etc/DependencyInstaller.sh
Comment (rovinski) [reply etc/DependencyInstaller.sh:136]: There would be nothing to resolve here, only in ORFS after this is merged.

File: etc/DependencyInstaller.sh
Comment (rovinski) [reply etc/DependencyInstaller.sh]: I realized this was likely affecting all builds so I removed the version requirement. It should only install on systems which don't have pcre2 at all.

File: etc/DependencyInstaller.sh
Comment (maliberty) [inline etc/DependencyInstaller.sh:56]: What happens in pcre2-config doesn't exist?

Comment (rovinski) [reply etc/DependencyInstaller.sh:56]: It downloads the tarball and uses swig's built-in script to build it and statically link it during the swig build.

**Review by oharboe — COMMENTED** [review]
Post-merge the ORFS env.sh and dev_env.sh should be changed to use the generated environment setup scripts in the installation to have a single source of truth.

**Review by oharboe — COMMENTED** [review]
Post-merge the ORFS env.sh and dev_env.sh should be changed to use the generated environment setup scripts in the installation to have a single source of truth.

**Review by oharboe — COMMENTED** [review]
Post-merge the ORFS env.sh and dev_env.sh should be changed to use the generated environment setup scripts in the installation to have a single source of truth.

**Comment by oharboe** [comment]
@rovinski I wasn't able to change my request for changes to a review comment, so I had to dismiss my own review with "Post-merge the ORFS env.sh and dev_env.sh should be changed to use the generated environment setup scripts in the installation to have a single source of truth."

**Comment by rovinski** [comment]
I raised the build errors as a separate issue in #3343

@maliberty anything else preventing this from merging?

**Comment by rovinski** [comment]
Closes #3343

**Comment by rovinski** [comment]
CentOS failures are unrelated. Network issues.

**Comment by rovinski** [comment]
I should note that before this PR, OpenSuse, testinstaller-macos, and rhel were all failing. Now, at least rhel is passing. If centos is rerun, it should pass (I don't have permissions to rerun github actions).

**Comment by vvbandeira** [comment]
I think it is OK to merge the PR as is; there are some changes coming that will make this more stable.

**Review by maliberty — APPROVED** [review]

**Pull request merged**
